app/src/components/market/sections/market_buy/scalar_market_buy.tsx

```diff
@@ -240,17 +281,25 @@ export const ScalarMarketBuy = (props: Props) => {
       if (!cpk) {
         return
       }
-
-      const sharesAmount = formatBigNumber(tradedShares, collateral.decimals)
-
+      let displayTradedShares = tradedShares
+      let useBaseToken = false
+      let inputAmount = amount || Zero
+      if (collateralSymbol in CompoundTokenType && compoundService && amount) {
+        displayTradedShares = compoundService.calculateCTokenToBaseExchange(baseCollateral, tradedShares)
+        useBaseToken = true
+        inputAmount = compoundService.calculateCTokenToBaseExchange(baseCollateral, amount)
+      }
+      const sharesAmount = formatBigNumber(displayTradedShares, baseCollateral.decimals)
       setStatus(Status.Loading)
       setMessage(`Buying ${sharesAmount} shares ...`)
 
       await cpk.buyOutcomes({
-        amount,
+        amount: inputAmount,
```

Comment:
One thing I found was that if I am buying off a cDai market with cDai as my selected currency, it will convert my Dai to cDai instead of using my existing cDai.

Reply:
Fixed